com.unity.ml-agents/Runtime/Inference/BarracudaModelParamLoader.cs

```diff
@@ -179,7 +248,88 @@ ISensor[] sensors
                     !tensorsNames.Any(x => x.EndsWith("_c")))
                 {
                     failedModelChecks.Add(
-                        "The model does not contain a Recurrent Input Node but has memory_size.");
+                        new FailedCheck
+                        {
+                            CheckType = CheckType.Warning,
+                            Message =
+                        "The model does not contain a Recurrent Input Node but has memory_size."
+                        });
+                }
+            }
+
+            // If the model uses discrete control but does not have an input for action masks
+            if (model.HasDiscreteOutputs())
+            {
+                if (!tensorsNames.Contains(TensorNames.ActionMaskPlaceholder))
+                {
+                    failedModelChecks.Add(
+                        new FailedCheck
+                        {
+                            CheckType = CheckType.Warning,
+                            Message =
+                        "The model does not contain an Action Mask but is using Discrete Control."
+                        });
+                }
+            }
+            return failedModelChecks;
+        }
+
+        /// <summary>
+        /// Generates failed checks that correspond to inputs expected by the model that are not
+        /// present in the BrainParameters.
+        /// </summary>
+        /// <param name="model">
+        /// The Barracuda engine model for loading static parameters
+        /// </param>
+        /// <param name="brainParameters">
+        /// The BrainParameters that are used verify the compatibility with the InferenceEngine
+        /// </param>
+        /// <param name="memory">
+        /// The memory size that the model is expecting.
+        /// </param>
+        /// <param name="sensors">Array of attached sensor components</param>
+        /// <returns>
+        /// A IEnumerable of string corresponding to the failed input presence checks.
+        /// </returns>
+        static IEnumerable<FailedCheck> CheckInputTensorPresence(
```

Comment:
Should you also check that the legacy names aren't here?

Reply:
All unknown names (including legacy ones) will be raised line 649 in CheckInputTensorShape.